crates/networking/rpc/engine/fork_choice.rs

```diff
@@ -4,72 +4,152 @@ use ethrex_blockchain::{
     latest_canonical_block_hash,
     payload::{create_payload, BuildPayloadArgs},
 };
+use ethrex_core::types::{BlockHeader, ChainConfig};
 use serde_json::Value;
 use tracing::{info, warn};
 
 use crate::{
     types::{
-        fork_choice::{ForkChoiceResponse, ForkChoiceState, PayloadAttributesV3},
+        fork_choice::{ForkChoiceResponse, ForkChoiceState, PayloadAttributes},
         payload::PayloadStatus,
     },
     utils::RpcRequest,
     RpcApiContext, RpcErr, RpcHandler,
 };
 
 #[derive(Debug)]
-pub struct ForkChoiceUpdatedV3 {
+pub enum ForkChoiceUpdatedVersion {
+    V1,
+    V2,
+    V3,
+}
+
+#[derive(Debug)]
+pub struct ForkChoiceUpdated {
     pub fork_choice_state: ForkChoiceState,
-    #[allow(unused)]
-    pub payload_attributes: Result<Option<PayloadAttributesV3>, String>,
+    pub payload_attributes: Result<Option<PayloadAttributes>, String>,
+    pub version: ForkChoiceUpdatedVersion,
 }
 
-impl TryFrom<ForkChoiceUpdatedV3> for RpcRequest {
-    type Error = String;
+impl ForkChoiceUpdated {
+    fn method(&self) -> String {
+        match self.version {
+            ForkChoiceUpdatedVersion::V1 => "engine_forkchoiceUpdatedV1".to_string(),
+            ForkChoiceUpdatedVersion::V2 => "engine_forkchoiceUpdatedV2".to_string(),
+            ForkChoiceUpdatedVersion::V3 => "engine_forkchoiceUpdatedV3".to_string(),
```

Comment:
The same trick as below enables formatting:
```suggestion
            format!("engine_forkchoiceUpdatedV{}", self.version as usize)
```